pkgs/by-name/ob/objdiff/package.nix

```
@@ -0,0 +1,63 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  darwin,
+}:
+let
+  version = "2.0.0-beta.6";
+  commit = "ce05d6d6c04a029ef2fcfa68ad25be8698789bdb";
+in
+rustPlatform.buildRustPackage {
+  pname = "objdiff";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "encounter";
+    repo = "objdiff";
+    rev = "v${version}";
+    hash = "sha256-o9Gqc04ZPVhU7I4Gz+r/m6c/6LwBZBrExkpGk5TlccA=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "notify-6.1.1" = "sha256-y3tbOIDWv9d2ZAQAo+xBJwzWcnYh7OO72izatfBK2QQ=";
+      "ppc750cl-0.3.0" = "sha256-nMJk+rgu7Ydi2VZfodJk0kBz9xLLVBVz0vEZPee8Q6M=";
+    };
+  };
+
+  buildInputs = lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.AppKit ];
+
+  prePatch = ''
+    # Don't use git, we don't have a .git. We'll pass the required data manually
+    # through environment variables.
+    substituteInPlace objdiff-gui/build.rs \
+      --replace-fail ".all_git()" ".disable_git()"
+
+    # objdiff-cli doesn't use vergen, but its own bespoke invocation of the git
+    # CLI. Remove the entire build script, and provide the data ourselves.
+    rm objdiff-cli/build.rs
+    substituteInPlace objdiff-cli/Cargo.toml \
+      --replace-fail 'build = "build.rs"' ""
+  '';
```

Non-blocker, but useful to avoid unnecessary rebuilds:

```suggestion
  prePatch = 
    # Don't use git, we don't have a .git. We'll pass the required data manually
    # through environment variables.
    ''
      substituteInPlace objdiff-gui/build.rs \
        --replace-fail ".all_git()" ".disable_git()"
    ''
    # objdiff-cli doesn't use vergen, but its own bespoke invocation of the git
    # CLI. Remove the entire build script, and provide the data ourselves.
    + ''
      rm objdiff-cli/build.rs
      substituteInPlace objdiff-cli/Cargo.toml \
        --replace-fail 'build = "build.rs"' ""
    '';
```

---

Fixed.